src/types/type_param.rs

```diff
@@ -70,43 +55,30 @@ impl TypeArg {
 /// Checks a [TypeArg] is as expected for a [TypeParam]
 pub fn check_type_arg(arg: &TypeArg, param: &TypeParam) -> Result<(), TypeArgError> {
     match (arg, param) {
-        (TypeArg::Type(_), TypeParam::Type) => Ok(()),
-        (TypeArg::ClassicType(_), TypeParam::ClassicType) => Ok(()),
-        (TypeArg::HashableType(_), TypeParam::HashableType) => Ok(()),
-        (TypeArg::List(items), TypeParam::List(ty)) => {
-            for item in items {
-                check_type_arg(item, ty.as_ref())?;
+        (
+            TypeArg::Type(SimpleType::Classic(ClassicType::Hashable(_))),
```

Comment:
Ah, just realized - I think you can do
```
(TypeArg::Type(t), TypeParam::Type(tag)) if tag.contains(t.tag()) => Ok(())
```
(You'll need `TypeTag::contains` but we should have that)